src/EditorFeatures/Test/Rename/RenameRewriterTests.Verifier.cs

```diff
@@ -129,36 +126,35 @@ public Verifier(string workspaceXml)
-                            var symbolContext = new RenamedSymbolContext(
-                                    replacementText,
-                                    renameSymbol.Name,
-                                    possibleNameConflicts,
-                                    renameSymbol,
-                                    renameSymbol as IAliasSymbol,
-                                    renameRewriterService.IsIdentifierValid(replacementText, syntaxFacts));
-
-                            renamedSymbolContextsBuilder.Add(symbolContext);
+                            documentRenameBuilder.AddRenamedSymbol(renameSymbol, replacementText, replacementTextValid, possibleNameConflicts.ToImmutableArray());
 
                             var renameLocationsSet = await Renamer.FindRenameLocationsAsync(
                                 solution,
                                 renameSymbol,
                                 options,
-                                CodeActionOptions.DefaultProvider,
                                 cancellationToken).ConfigureAwait(false);
 
                             var locationsInDocument = renameLocationsSet.Locations.Where(location => location.DocumentId == document.Id).ToImmutableArray();
-                            var textSpanRenameContexts = locationsInDocument
+                            var locationRenameContexts = locationsInDocument
                                 .WhereAsArray(location => RenameUtilities.ShouldIncludeLocation(renameLocationsSet.Locations, location))
-                                .SelectAsArray(location => new LocationRenameContext(location, symbolContext));
-                            tokenTextSpanRenameContextsBuilder.AddRange(textSpanRenameContexts);
+                                .SelectAsArray(location => new LocationRenameContext(location, replacementTextValid, replacementText, renameSymbol.Name));
+
+                            foreach (var locationRenameContext in locationRenameContexts)
+                            {
+                                documentRenameBuilder.AddLocationRenameContext(locationRenameContext);
+                            }
 
                             var stringAndCommentsRenameContexts = locationsInDocument
                                 .WhereAsArray(location => location.IsRenameInStringOrComment)
-                                .SelectAsArray(location => new LocationRenameContext(location, symbolContext));
-                            stringAndCommentsContextsBuilder.AddRange(stringAndCommentsRenameContexts);
+                                .SelectAsArray(location => new StringAndCommentRenameContext(location, replacementText));
+                            foreach (var stringAndCommentRenameContext in stringAndCommentsRenameContexts)
+                            {
+                                documentRenameBuilder.AddStringAndCommentRenameContext(stringAndCommentRenameContext);
```

Comment:
nit: Why not just have a range overload for this? Makes the code slightly cleaner here